akka-bbb-apps/src/main/scala/org/bigbluebutton/core/models/Polls.scala

```diff
@@ -0,0 +1,619 @@
+
+import scala.collection.mutable.ArrayBuffer
+import scala.collection.mutable.HashMap
+import org.bigbluebutton.core.running.{ LiveMeeting, MeetingActor }
+
+object Polls {
+
+  def handleStartPollReqMsg(userId: String, pollId: String, pollType: String, lm: LiveMeeting): Option[SimplePollOutVO] = {
+    println("Received StartPollReqMsg for pollType=[" + pollType + "]")
+    def createPoll(pollId: String, numRespondents: Int): Option[Poll] = {
+      for {
+        poll <- PollFactory.createPoll(pollId, pollType, numRespondents, None)
+      } yield {
+        lm.polls.save(poll)
+        poll
+      }
+    }
+
+    for {
+      page <- lm.presModel.getCurrentPage()
+      pageId: String = if (pollId.contains("deskshare")) "deskshare" else page.id
+      stampedPollId: String = pageId + "/" + System.currentTimeMillis() // TODO - is the timestamp needed?
+      numRespondents: Int = Users.numUsers(lm.users) - 1 // subtract the presenter
+
+      poll <- createPoll(stampedPollId, numRespondents)
+      simplePoll <- getSimplePoll(poll.id, lm.polls)
+    } yield {
+      startPoll(simplePoll.id, lm.polls)
+      simplePoll
+    }
+  }
+
+  def handleStopPollReqMsg(userId: String, lm: LiveMeeting): Option[String] = {
+    for {
+      page <- lm.presModel.getCurrentPage()
+      curPoll <- getRunningPollThatStartsWith(page.id, lm.polls)
+    } yield {
+      stopPoll(curPoll.id, lm.polls)
+      curPoll.id
+    }
+  }
+
+  def handleShowPollResultReqMsg(requesterId: String, pollId: String, lm: LiveMeeting): Option[SimplePollResultOutVO] = {
+    //    def send(poll: SimplePollResultOutVO, shape: scala.collection.immutable.Map[String, Object]): Unit = {
+    //      for {
+    //        page <- lm.presModel.getCurrentPage()
+    //        pageId = if (poll.id.contains("deskshare")) "deskshare" else page.id
+    //        annotation = new AnnotationVO(poll.id, WhiteboardKeyUtil.DRAW_END_STATUS, WhiteboardKeyUtil.POLL_RESULT_TYPE, shape, pageId, requesterId, -1)
+    //      } handleSendWhiteboardAnnotationRequest(new SendWhiteboardAnnotationRequest(props.meetingProp.intId, requesterId, annotation))
+    //    }
+
+    for {
+      result <- getSimplePollResult(pollId, lm.polls)
+      shape = pollResultToWhiteboardShape(result)
+    } yield {
+      //      send(result, shape)
+      showPollResult(pollId, lm.polls)
+      // outGW.send(new PollShowResultMessage(props.meetingProp.intId, props.recordProp.record, msg.requesterId, msg.pollId, result))
+      result
+    }
+  }
+
+  def handleHidePollResultReqMsg(requesterId: String, pollId: String, lm: LiveMeeting): Option[String] = {
+    for {
+      poll <- getPoll(pollId, lm.polls)
+    } yield {
+      hidePollResult(pollId, lm.polls)
+      // outGW.send(new PollHideResultMessage(props.meetingProp.intId, props.recordProp.record, requesterId, pollId))
+      pollId
+    }
+  }
+
+  def handleGetCurrentPollReqMsg(requesterId: String, lm: LiveMeeting): (Boolean, Option[PollVO]) = {
```

Comment:
Couldn't you just return the Option? Some means it exists and None means it doesn't